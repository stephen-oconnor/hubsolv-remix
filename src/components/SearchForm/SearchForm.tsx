import React from 'react';
import PropTypes from 'prop-types';
import SearchInput from '../common/forms/SearchInput';
import IconSearch from '../Icons/IconSearch';
import s from './styles.module.scss';

export interface SearchFormProps {
	value?: string | undefined;
	loading: boolean;
	error: boolean;
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
	onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
}

const SearchForm: React.FC<SearchFormProps> = ({
	value,
	loading = false,
	error = false,
	onChange,
	onSubmit,
}) => {
	return (
		<form className={s.form} onSubmit={onSubmit}>
			<div className={s.item}>
				<SearchInput name={'search'} label={'Search'} onChange={onChange} value={value} />
			</div>
			<button type="submit" className={s.submit}>
				{<span className={s.srt}>{loading ? 'Loading...' : 'Search'}</span>}
				{<IconSearch width={24} height={24} fill={'#001724'} />}
			</button>
		</form>
	);
};

SearchForm.propTypes = {
	value: PropTypes.string,
	loading: PropTypes.bool.isRequired,
	error: PropTypes.bool.isRequired,
	onChange: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
