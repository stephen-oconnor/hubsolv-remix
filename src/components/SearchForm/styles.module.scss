@import '../../styles/base/config.scss';

.form {
	display: flex;
	padding: 0.7rem;
	background-color: rgba(255, 255, 255, 0.2);
	border-radius: 5px;
}

.item {
	display: flex;
	flex: 1;
}

.submit {
	display: flex;
	padding: 1rem;
	color: $colour_secondary;
	background-color: $colour_tertiary;
	border: 0;
	border-top-right-radius: 5px;
	border-bottom-right-radius: 5px;
	font-size: 1.5rem;
	outline: none;
}

.srt {
	border: 0;
	clip: rect(1px 1px 1px 1px);
	clip: rect(1px, 1px, 1px, 1px);
	height: 1px;
	margin: -1px;
	overflow: hidden;
	padding: 0;
	position: absolute;
	width: 1px;
}

@media screen and (min-width: 768px) {
	.form {
		max-width: 600px;
		margin: 0 auto;
	}
}
