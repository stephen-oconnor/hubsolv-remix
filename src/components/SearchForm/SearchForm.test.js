import React from 'react';
import SearchForm from './SearchForm';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<SearchForm />', () => {
	const onSubmitMock = jest.fn();
	const props = {
		loading: false,
		errors: {},
		onChange: jest.fn(),
		onSubmit: onSubmitMock,
	};

	const event = {
		// eslint-disable-next-line @typescript-eslint/no-empty-function
		preventDefault() {},
	};

	it('matches the snapshot', () => {
		const tree = shallow(<SearchForm {...props} />);
		expect(toJson(tree)).toMatchSnapshot();
	});

	it('should call on submit when form is submitted, not submit the form', () => {
		const wrapper = shallow(<SearchForm {...props} />);

		wrapper.find('form').simulate('submit', event);

		expect(onSubmitMock).toBeCalledWith(event);
	});
});
