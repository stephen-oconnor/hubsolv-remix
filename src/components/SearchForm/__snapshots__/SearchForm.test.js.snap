// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> matches the snapshot 1`] = `
<form
  className="form"
  onSubmit={[MockFunction]}
>
  <div
    className="item"
  >
    <SearchInput
      label="Search"
      name="search"
      onChange={[MockFunction]}
    />
  </div>
  <button
    className="submit"
    type="submit"
  >
    <span
      className="srt"
    >
      Search
    </span>
    <IconSearch
      fill="#001724"
      height={24}
      width={24}
    />
  </button>
</form>
`;
